src/components/list-item/daikin-list-item.ts

```diff
@@ -0,0 +1,183 @@
+    "focus-visible:outline-daikinBlue-700",
+  ],
+  {
+    variants: {
+      leftIcon: {
+        false: ["pl-4"],
+        true: ["pl-3"],
+      },
+      rightIcon: {
+        false: ["pr-4"],
+        true: ["pr-3"],
+      },
+    },
+  }
+);
+
+/**
+ * The list item component functions as a child element of the list component, and is used to actually list items.
+ *
+ * You can choose between the `button` type, which uses the HTML button tag, and the `link` type, which uses the a tag, and use them according to your needs.
+ *
+ * Hierarchy:
+ * - `daikin-list` > `daikin-list-item`
+ *
+ * @fires click - A retargeted event of a [click event](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event). Suppressed if `disabled` is true.
+ *
+ * @slot - An optional element that is displayed on the right of a list item. e.g. `daikin-checkbox`, `daikin-toggle`
+ *
+ * @example
+ *
+ * ```html
+ * <daikin-list-item label="List item label"></daikin-list-item>
+ * ```
+ */
+@customElement("daikin-list-item")
+export class DaikinListItem extends LitElement {
+  static override readonly styles = css`
+    ${unsafeCSS(tailwindStyles)}
+  `;
+
+  /**
+   * Label text for the list item.
+   */
+  @property({ type: String, reflect: true })
+  label = "";
+
+  /**
+   * Type of the list item.
+   * If `"link"` is specified, the list item will be rendered as an `<a>` element or `<span>` element (if `disabled` is `true`).
+   */
+  @property({ type: String, reflect: true })
+  type: "button" | "link" = "button";
+
+  /**
+   * Link `href`.
+   * Only used if the `type` is `"link"`.
+   */
+  @property({ type: String, reflect: true })
+  href: string | null = null;
+
+  /**
+   * An icon displayed at the left of the label.
+   */
+  @property({ type: String, reflect: true, attribute: "left-icon" })
+  leftIcon: IconType | null = null;
+
+  /**
+   * Whether the right arrow icon is visible.
+   * If there is content in the slot, it will always be false.
+   */
+  @property({ type: Boolean, reflect: true, attribute: "right-arrow-icon" })
+  rightArrowIcon: boolean = false;
+
+  /**
+   * Whether the list item is disabled.
+   */
+  @property({ type: Boolean, reflect: true })
+  disabled: boolean = false;
+
+  @query("a,button")
+  private _focusableElement!: HTMLAnchorElement | HTMLButtonElement | null;
+
+  constructor() {
+    super();
+
+    this.addEventListener("click", (event: MouseEvent): void => {
+      if (this.disabled) {
+        event.stopImmediatePropagation();
+      }
+    });
+  }
+
+  override render() {
+    const listCN = cvaListItem({
+      leftIcon: !!this.leftIcon,
+      rightIcon: this.rightArrowIcon,
+    });
+
+    const leftContent = html`<span class="flex items-center flex-none gap-2">
+      ${this.leftIcon
+        ? html`<daikin-icon
+            icon=${this.leftIcon}
+            size="xl"
+            color="current"
+          ></daikin-icon>`
+        : nothing}
+      ${this.label}
```

Comment:
Please use slots for everything that is rendered on the DOM. In particular, please use the default slot for the main content.
If you need additional slots, please use named slots.

```suggestion
      <slot></slot>
```